### ISSUE TITLE
Feature Request: Widget cycling

### ISSUE THREAD

**Issue by enaloha** [KpwnZ/Bug-report-feature-request-for-SimpleLS-2#16]
Not sure if SimpleLS can detect taps, but here's a feature that would significantly increase its utility: allow tapping the widget area to cycle through the widgets enabled in the Settings. So the weather widget could potentially show the original info with temperature and, if enabled, a tap on the widget would advance to sunrise/sunset (pne could even consider adding more weather infos. A further tap would advance through any other enabled widgets.

**Comment by KpwnZ** [comment]
Great idea! Thanks for your support. I will try to add it to future version.

**Comment by enaloha** [comment]
Your responsiveness is amazing. Thank you so much for making this wonderful utility available and even taking feature requests. SimpleLS is one the most useful utilities for the Lockscreen. Very nice implementation.

If you are indeed going to implement the widget cycling, I would like to add another idea for the Weather widget, which would absolutely delight any person in the US who is in contact with people abroad or vice versa. Only a few countries in the world use Fahrenheit as their official scale: the United States, Belize, Palau, the Bahamas and the Cayman Islands. All others use Celsius. So if you want to talk about temperatures with people not familiar with your particular scale, you constantly have to convert. I am not aware of any utility that will make this process easier. May be SimpleLS could come to the rescue by offering this option. Here are two ways in wich this this could be accomplished:

1. Have a setting that enables the standard weather widget to show the Default (determined by the Weather.app settingl let's say °F) plus the alternative °C. If disabled, the weather would be shown as it is now (**City • 77°**). If enabled, the widget would show both (**City • 77°F - 25°C**). This option could be set in the Settings -> Widget Options -> Weather

2. An alternative implementation could be to allow alternating temperature scales within the standard weather widget. Similar to option 1 above, the temperature would be as it is now, but with the unit appended (**City • 77°F**). If enabled, the widget would automatically alternate between Fahrenheit and Celsius (one second (**City • 77°F**) followed by one second (**City • 25°C**). This option could be set in the Settings -> Widget Options -> Weather as above. If you want to get fancy, you could even let the user choose which version they want: "Show both °F and °C" or "Alternate °F and °C"

3. Finally, one could implement this feature as part of the widget cycling. Currently, the weather-related widgets are the standard Weather and the Sunrise/Sunset widgets. One could simply create two standard Weather widgets, one would be "Weather °F" and the other would be "Weather °C". If cycling widgets is implemented, the user could simply enable both and switch between °F and °C by tapping on the widget. This implementation would be similar to the auto-cycling option 1 above, just using manual switching.

These are just suggestions that would let me better communicate with my relatives in Europe who only know Celsius, whereas I am forced to use Fahrenheit in the US. I am sure that there many multi-national iphone users who would love to see this feature.

**Comment by GeoDoX** [comment]
Would work very well with #18 too.

**Comment by enaloha** [comment]
Indeed, widget cycling and additional useful widgets would make SimpleLS an indispensable utility.

**Comment by enaloha** [comment]
IT LIVE in the latest beta. This is really awesome.

**Comment by enaloha** [comment]
I should qualify that as far as I can tell, the cycling is currently limited to the Weather widget (switching between weather conditions and sunrise/sunset). Tapping on the weather will switch between the two.

**Comment by KpwnZ** [comment]
> I should qualify that as far as I can tell, the cycling is currently limited to the Weather widget (switching between weather conditions and sunrise/sunset). Tapping on the weather will switch between the two.

It will be able to cycling between all widgets later.

**Comment by KpwnZ** [comment]
> I should qualify that as far as I can tell, the cycling is currently limited to the Weather widget (switching between weather conditions and sunrise/sunset). Tapping on the weather will switch between the two.

Version 1.3.0-b2 can switch between all widgets now.

**Comment by enaloha** [comment]
That's cool and it's working for me on iPhone 7 plus iOS 12.4 just fine. Would it be possible to change the tweak so that only enabled widgets cycle? One way to achieve that is to not use a switch to enable cycling (as is implemented now), but instead allow the user to select more than one widget. If only one widget is checked, it means "no cycling". If you check a second widget, it automatically means you want cycling and this way, the user can check those widgets that he/she wants cycling.

**Comment by enaloha** [comment]
sorry, clicked the wrong button (did not intend to close)
